library/alloc/src/vec.rs

```diff
@@ -2730,6 +2730,26 @@ impl<T> Iterator for IntoIter<T> {
     fn count(self) -> usize {
         self.len()
     }
+
+    #[inline]
+    fn last(mut self) -> Option<T> {
+        unsafe {
+            if self.ptr as *const _ == self.end {
+                None
+            } else {
+                if mem::size_of::<T>() == 0 {
+                    // Immediately marches to end of the iterator.
+                    self.ptr = self.end;
+                    // Make up a value of this ZST.
+                    Some(mem::zeroed())
+                } else {
+                    // Immediately marches to end of the iterator.
+                    self.ptr = self.end;
+                    Some(ptr::read(self.ptr.offset(-1)))
+                }
+            }
+        }
```

Comment:
```suggestion
        self.next_back()
```

No need to reinvent the wheel here.

Reply:
No, this does not consume the iterator. `next_back` only modifies `self.end` which `next` will still return something.

Reply:
`last` takes `self` by value so the iterator is consumed and it's not possible to call `next` afterwards.

Reply:
> `next_back` only modifies `self.end`

That's what we need. If you do `self.ptr = self.end;` then it will leak all of the other items when `IntoIter` is dropped.

Reply:
Ah, I didn't know the leak you all mentioned is because `self.ptr = self.end;`.